packages/react-components/react-swatch-picker-preview/docs/Spec.md

```diff
@@ -0,0 +1,290 @@
+
+### States
+
+- `hover`
+- `selected`
+- `focused`
+- `pressed`
+
+## API
+
+## Structure
+
+### Components
+
+| Component    | Purpose                                                |
+| ------------ | ------------------------------------------------------ |
+| SwatchPicker | Group which can represent swatches as a row or a grid. |
+| ColorSwatch  | Swatch for a color                                     |
+| ImageSwatch  | Swatch for an image, texture or a pattern              |
+
+## SwatchPicker component
+
+#### Anatomy
+
+![visual anatomy of the SwatchPicker component](./assets/todo.png)
+
+#### DOM
+
+```HTML
+<div role="radiogroup" aria-labelledby="colors">
+  {children}
+</div>
+```
+
+#### ColorPicker structure
+
+- root `div` element
+
+#### API
+
+| Property | Values                                   | Default  | Purpose                         |
+| -------- | ---------------------------------------- | -------- | ------------------------------- |
+| shape    | `square`, `circular`, `rounded`          | `square` | Sets shape                      |
+| size     | `extraSmall`, `small`, `medium`, `large` | `medium` | Defines size of the Swatch cell |
+| id       | number                                   |          | Sets swatch id                  |
+| layout   | `row`, `grid`                            | `grid`   | Sets layout of the SwatchPicker |
+| onChange |                                          |          |                                 |
+
+## ColorSwatch component
+
+is used for picking colors:
+
+- solid color
+- gradient
+
+#### Anatomy
+
+![visual anatomy of the SwatchPicker component](./assets/todo.png)
+
+#### DOM
+
+```HTML
+<span name="color" value="{color-id}" style="background: red;">
+  <input type="radio" id="{color-id}" name="color" value="red">
+</span>
+<span name="color" value="{color-id}" style="background: rgb(189, 255, 104);">
+  <input type="radio" id="{color-id}" name="color" value="{color-id}">
+</span>
+```
+
+#### ColorSwatch structure
+
+- root `span` element
+- `input` element
+
+#### API
+
+| Property | Values                                   | Default  | Purpose                         |
+| -------- | ---------------------------------------- | -------- | ------------------------------- |
+| shape    | `square`, `circular`, `rounded`          | `square` | Sets shape                      |
+| size     | `extraSmall`, `small`, `medium`, `large` | `medium` | Defines size of the Swatch cell |
+| id       | number                                   |          | Sets swatch id                  |
+| name     |                                          |          | Sets color id                   |
+| swatch   |                                          |          | Sets color id                   |
+| disabled | boolean                                  |          |                                 |
+| selected | boolean                                  |          |                                 |
```

Comment:
If it's an `input`, should not it be `checked`?

Reply:
not input anymore